flood_adapt/object_model/scenario.py

```diff
@@ -57,13 +57,34 @@ def save(self, filepath: Union[str, os.PathLike]):
     def run(self):
         """run direct impact models for the scenario"""
         self.init_object_model()
+        # preprocess model input data first
         if not self.direct_impacts.hazard.has_run:
-            self.direct_impacts.hazard.run_models()
+            self.direct_impacts.hazard.preprocess_models()
         else:
             print(f"Hazard for scenario '{self.attrs.name}' has already been run.")
+        if not self.direct_impacts.has_run:
+            # self.direct_impacts.preprocess_models() #TODO: separate preprocessing and running of impact models
+            ...
+        else:
+            print(
+                f"Direct impacts for scenario '{self.attrs.name}' has already been run."
+            )
+
+        if not self.direct_impacts.hazard.has_run:
+            self.direct_impacts.hazard.run_models()
         if not self.direct_impacts.has_run:
             self.direct_impacts.run_models()
         else:
             print(
                 f"Direct impacts for scenario '{self.attrs.name}' has already been run."
             )
+
+        if not self.direct_impacts.hazard.has_run:
+            self.direct_impacts.hazard.postprocess_models()
```

Comment:
isn't this needed before you run direct impacts? At least for the risk run where the RP flood maps are needed as input

Reply:
I included all functions in one if statement: 

        if not self.direct_impacts.hazard.has_run:
            self.direct_impacts.hazard.preprocess_models()
            self.direct_impacts.hazard.run_models()
            self.direct_impacts.hazard.postprocess_models()
        else:
            print(f"Hazard for scenario '{self.attrs.name}' has already been run.")